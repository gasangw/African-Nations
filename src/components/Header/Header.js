import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import Container from 'react-bootstrap/Container';
import Navbar from 'react-bootstrap/Navbar';
import Form from 'react-bootstrap/Form';
import Button from 'react-bootstrap/Button';
import { FaLessThan, FaMicrophone } from 'react-icons/fa';
import { FiSettings } from 'react-icons/fi';
import { Link } from 'react-router-dom';
import { searchCountry } from '../../redux/countries/countries';

const Header = () => {
  const [search, setSearch] = useState('');
  const dispatch = useDispatch();
  const handleSearch = (e) => {
    setSearch(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (search.length > 0) {
      dispatch(searchCountry(search));
    }
  };
  return (
    <Navbar>
      <Container>
        <Link to="/">
          <Navbar.Brand className="year">
            <FaLessThan className="icon" />
            {' '}
            <span>2022</span>
          </Navbar.Brand>
        </Link>
        <Form className="d-flex justify-content-end flex-grow-1 pe-3" onSubmit={handleSubmit}>
          <Form.Control
            type="search"
            placeholder="Search"
            className="me-2"
            aria-label="Search"
            value={search}
            name="country"
            onChange={handleSearch}
          />
          <Button variant="outline-success" type="submit">Search</Button>
        </Form>
        <Navbar.Collapse className="justify-content-end">
          <Navbar.Text>
            <FaMicrophone className="icon" />
          </Navbar.Text>
          <Navbar.Text>
            <FiSettings className="icon" />
          </Navbar.Text>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};

export default Header;
