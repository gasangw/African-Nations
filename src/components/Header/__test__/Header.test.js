import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../../redux/configureStore';
import Header from '../Header';

describe('Test header', () => {
  it('test if the heander is rendered correctly', () => {
    const header = renderer
      .create(
        <Provider store={store}>
          <Router>
            <Header />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(header).toMatchSnapshot();
  });
});
