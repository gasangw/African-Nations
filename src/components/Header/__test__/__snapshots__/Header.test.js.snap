// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test header test if the heander is rendered correctly 1`] = `
<nav
  className="navbar navbar-expand navbar-light"
>
  <div
    className="container"
  >
    <a
      href="#/"
      onClick={[Function]}
    >
      <span
        className="year navbar-brand"
      >
        <svg
          className="icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 384 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M365.46 357.74L147.04 255.89l218.47-101.88c16.02-7.47 22.95-26.51 15.48-42.53l-13.52-29C360 66.46 340.96 59.53 324.94 67L18.48 209.91a32.014 32.014 0 0 0-18.48 29v34.24c0 12.44 7.21 23.75 18.48 29l306.31 142.83c16.06 7.49 35.15.54 42.64-15.52l13.56-29.08c7.49-16.06.54-35.15-15.53-42.64z"
          />
        </svg>
         
        <span>
          All
        </span>
      </span>
    </a>
    <form
      className="d-flex justify-content-end flex-grow-1 pe-3"
      onSubmit={[Function]}
    >
      <input
        aria-label="Search"
        className="me-2 form-control"
        name="country"
        onChange={[Function]}
        placeholder="Search your favorite country"
        type="search"
        value=""
      />
      <button
        className="btn btn-outline-success"
        disabled={false}
        type="submit"
      >
        Search
      </button>
    </form>
    <div
      aria-expanded={null}
      className="justify-content-end navbar-collapse collapse"
    >
      <span
        className="navbar-text"
      >
        <svg
          className="icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 352 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M176 352c53.02 0 96-42.98 96-96V96c0-53.02-42.98-96-96-96S80 42.98 80 96v160c0 53.02 42.98 96 96 96zm160-160h-16c-8.84 0-16 7.16-16 16v48c0 74.8-64.49 134.82-140.79 127.38C96.71 376.89 48 317.11 48 250.3V208c0-8.84-7.16-16-16-16H16c-8.84 0-16 7.16-16 16v40.16c0 89.64 63.97 169.55 152 181.69V464H96c-8.84 0-16 7.16-16 16v16c0 8.84 7.16 16 16 16h160c8.84 0 16-7.16 16-16v-16c0-8.84-7.16-16-16-16h-56v-33.77C285.71 418.47 352 344.9 352 256v-48c0-8.84-7.16-16-16-16z"
          />
        </svg>
      </span>
      <span
        className="navbar-text"
      >
        <svg
          className="icon"
          fill="none"
          height="1em"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="12"
            cy="12"
            r="3"
          />
          <path
            d="M19.4 15a1.65 1.65 0 0 0 .33 1.82l.06.06a2 2 0 0 1 0 2.83 2 2 0 0 1-2.83 0l-.06-.06a1.65 1.65 0 0 0-1.82-.33 1.65 1.65 0 0 0-1 1.51V21a2 2 0 0 1-2 2 2 2 0 0 1-2-2v-.09A1.65 1.65 0 0 0 9 19.4a1.65 1.65 0 0 0-1.82.33l-.06.06a2 2 0 0 1-2.83 0 2 2 0 0 1 0-2.83l.06-.06a1.65 1.65 0 0 0 .33-1.82 1.65 1.65 0 0 0-1.51-1H3a2 2 0 0 1-2-2 2 2 0 0 1 2-2h.09A1.65 1.65 0 0 0 4.6 9a1.65 1.65 0 0 0-.33-1.82l-.06-.06a2 2 0 0 1 0-2.83 2 2 0 0 1 2.83 0l.06.06a1.65 1.65 0 0 0 1.82.33H9a1.65 1.65 0 0 0 1-1.51V3a2 2 0 0 1 2-2 2 2 0 0 1 2 2v.09a1.65 1.65 0 0 0 1 1.51 1.65 1.65 0 0 0 1.82-.33l.06-.06a2 2 0 0 1 2.83 0 2 2 0 0 1 0 2.83l-.06.06a1.65 1.65 0 0 0-.33 1.82V9a1.65 1.65 0 0 0 1.51 1H21a2 2 0 0 1 2 2 2 2 0 0 1-2 2h-.09a1.65 1.65 0 0 0-1.51 1z"
          />
        </svg>
      </span>
    </div>
  </div>
</nav>
`;
